pkg/webhook/scheduler/mutating/schedule_pod_handler.go

```diff
@@ -139,7 +140,11 @@ func (a *CreateUpdatePodForSchedulingHandler) AddScheduleInfoToPod(pod *corev1.P
 	// handle the pods interact with fluid
 	switch {
 	case utils.ServerlessEnabled(pod.GetLabels()):
-		pluginsList = pluginsRegistry.GetServerlessPodHandler()
+		if len(runtimeInfos) == 0 {
+			pluginsList = pluginsRegistry.GetServerlessPodWithoutDatasetHandler()
+		} else {
+			pluginsList = pluginsRegistry.GetServerlessPodWithDatasetHandler()
```

Comment:
@xliuqq could you add unit test for line #L146 which was not covered by tests?

Reply:
ok, fixed.